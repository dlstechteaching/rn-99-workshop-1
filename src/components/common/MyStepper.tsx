import { View, StyleSheet } from 'react-native';

interface MyStepperProps {
    stepIndex: number;
    nbOfSteps: number;
    width?: string;
}

export const MyStepper: React.VFC<MyStepperProps> = ({ stepIndex, nbOfSteps, width }) => {

	return (
		<View style={[styles.container, { width: width ?? '100%'}]}>
			{
				Array.from(Array(nbOfSteps), (_, index: number) => {
					return <View key={index} style={[ styles.bullet, stepIndex === index ? styles.selected : undefined]} />;
				})
			}
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flexDirection: 'row',
		justifyContent: 'space-around',
		alignItems: 'center',
	},
	bullet: {
		height: 15,
		width: 15,
		borderRadius: 15,
		backgroundColor: 'rgb(239, 183, 114)'
	},
	selected: {
		backgroundColor: 'rgb(81, 53, 81)',
		height: 20, 
		width: 20,
		borderColor: 'rgb(98, 62, 97)',
		borderWidth: 1
	}
});