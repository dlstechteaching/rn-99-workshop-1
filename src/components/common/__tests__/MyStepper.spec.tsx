import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { MyStepper } from '../MyStepper';

describe('MyStepper tests', () => {
	let component: ShallowWrapper;

	it('should render properly', () => {
		givenComponent();
		thenItRendersProperly();
	});

	function givenComponent() {
		component = shallow(<MyStepper stepIndex={0} nbOfSteps={4} />);
	}

	function thenItRendersProperly() {
		expect(component).toMatchSnapshot();
	}
});
