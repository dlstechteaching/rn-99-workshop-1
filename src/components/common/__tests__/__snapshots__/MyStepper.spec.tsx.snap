// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyStepper tests should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
      Object {
        "width": "100%",
      },
    ]
  }
>
  <View
    key="0"
    style={
      Array [
        Object {
          "backgroundColor": "rgb(239, 183, 114)",
          "borderRadius": 15,
          "height": 15,
          "width": 15,
        },
        Object {
          "backgroundColor": "rgb(81, 53, 81)",
          "borderColor": "rgb(98, 62, 97)",
          "borderWidth": 1,
          "height": 20,
          "width": 20,
        },
      ]
    }
  />
  <View
    key="1"
    style={
      Array [
        Object {
          "backgroundColor": "rgb(239, 183, 114)",
          "borderRadius": 15,
          "height": 15,
          "width": 15,
        },
        undefined,
      ]
    }
  />
  <View
    key="2"
    style={
      Array [
        Object {
          "backgroundColor": "rgb(239, 183, 114)",
          "borderRadius": 15,
          "height": 15,
          "width": 15,
        },
        undefined,
      ]
    }
  />
  <View
    key="3"
    style={
      Array [
        Object {
          "backgroundColor": "rgb(239, 183, 114)",
          "borderRadius": 15,
          "height": 15,
          "width": 15,
        },
        undefined,
      ]
    }
  />
</View>
`;
